### ISSUE TITLE
create_sibling_github erroring with `existing=skip` when sibling exists

### ISSUE THREAD

**Issue by jwodder** [datalad/datalad#6945]
### What is the problem?

When calling `ds.create_sibling_github()` with `existing="skip"`, if the dataset does not have a sibling configured and the would-be sibling already exists on GitHub, an error results with the message "A repository '%s' already exists at '%s', use existing=reconfigure to use it as a sibling".

### DataLad information

datalad 0.17.2 on Debian 11 (bullseye)

**Comment by mih** [comment]
I was able to replicate this behavior. Although OP does not state it explicitly, I assume the expected behavior is that is actually skips this repo. This would imply yielding a `notneeded` result. That in turn would be confusing, because the outcome of `create-sibling...` is normally a configured remote, which would NOT be the case, despite a `notneeded` result.

It would be good to clarify the intention of this issue. Thx!

**Comment by jwodder** [comment]
@mih

> Although OP does not state it explicitly, I assume the expected behavior is that is actually skips this repo.

Something like that, yes.

> This would imply yielding a `notneeded` result. That in turn would be confusing

What exactly is `existing="skip"` supposed to do in the first place if skipping isn't considered an option?

CC @yarikoptic, who indicated that the current behavior is a bug in chat.

**Comment by yarikoptic** [comment]
TL;DR: I think I have misread the situation. @jwodder -- please do change to `='reconfigure'` from `='skip'`, but retaining the check that remote is not yet known.

The root of the issue seems to be in `create_sibling_github` being a tandem of 
1. create repo on github
2. add it as a remote.

`--existing` somewhat assumes both 1. and 2. but based solely on 2. (presence of the remote).  @jwodder ran into the use-case where `1.` has happened (might have been an independent actor).  `--existing=reconfigure` would also be not entirely correct setting unconditionally, i.e. we could not just set it since it would keep reconfiguring a possibly existing remote added in step 2.  But in https://github.com/dandi/dandisets/blob/HEAD/tools/backups2datalad/adataset.py#L212 we already condition on `"github"` remote not being known. So if we run into the case where 1. was done and 2. not (no remote) -- indeed we could just try to "reconfigure".  Overall it might work in the subsequent `push` or not depending on either that repository on github has history which local repository does not.

**Comment by yarikoptic** [comment]
as such as I do not think we should actually do anything on datalad side, since it would require decoupling singular --existing into two for separate 1 and 2. aspects, and IMHO too much to be worthwhile
